yomigeek/mtracker-app
# #157753167 User Signup API

#### What does this PR do?
* Create an API endpoint to for a user to signup

#### Description of Task to be completed?
* User should be able to create an account through this endpoint

#### How should this be manually tested?
* Create a new project folder on your computer 
* Go to the created folder in your terminal depending on your computer operating system
* Clone the project to your local repository from git clone https://github.com/yomigeek/mtracker-app.git
* All necessary dependencies as stated in the README.md file should be installed
* run $ npm install
* Run app locally using $ npm run start
* Open the postman application and enter the URL http://localhost:5000/api/v1/auth/signup with POST request method selected and also enter the required fields such as fullname, password, department, email
```
{
	"fullname": "Yomi user",
	"email":  "user@gmail.com",
	"password": "11111",
	"department": "technology"
	
}
```
* After the above is successful, you should get the following response: 
```
{
    "status": "success",
    "message": "User created successfully!",
    "data": {
	"fullname": "Yomi user",
	"email":  "user@gmail.com",
	"department": "technology"
    }
}
```
#### Any background context you want to provide?
All  fields are important i.e fullname, password, department, email must be supplied

####  What are the relevant pivotal tracker stories?
story type: feature
story id: 157753167

#### What are the relevant Github Issues?
None

#### Questions:
None
